src/parser/ast_node/ast_node.rs

```diff
@@ -0,0 +1,47 @@
+// #[derive(Debug)]
+use super::ast_node_concat::AstNodeConcat;
+use super::ast_node_group::AstNodeGroup;
+use super::ast_node_literal::AstNodeLiteral;
+use super::ast_node_optional::AstNodeOptional;
+use super::ast_node_plus::AstNodePlus;
+use super::ast_node_star::AstNodeStar;
+use super::ast_node_union::AstNodeUnion;
+
+pub(crate) enum AstNode {
+    Literal(AstNodeLiteral),   // Single character literal
+    Concat(AstNodeConcat),     // Concatenation of two expressions
+    Union(AstNodeUnion),       // Union of two expressions
+    Star(AstNodeStar),         // Kleene Star (zero or more)
+    Plus(AstNodePlus),         // One or more
+    Optional(AstNodeOptional), // Zero or one (optional)
+    Group(AstNodeGroup),       // Capturing group
```

Comment:
```suggestion
    Literal(AstNodeLiteral),
    Concat(AstNodeConcat),
    Union(AstNodeUnion),
    Star(AstNodeStar),
    Plus(AstNodePlus),
    Optional(AstNodeOptional),
    Group(AstNodeGroup),
```
Ideally these comments should be the doc string of the actual variant. I'd prefer to remove them to make the enum definition cleaner